### PR TITLE
Switch from gometalinter to golangci-lint

### PR DESCRIPTION
**Pull request by mjtrangoni**
Hi @torkelo @xlson @bergquist ,

I found [golangci-lint](https://github.com/golangci/golangci-lint), which is on average [5 times faster](https://github.com/golangci/golangci-lint#performance) than [gometalinter](https://github.com/alecthomas/gometalinter).

This fixes some issues found by **golangci-lint**, mostly **goconst** issues which are trivial.
I leave **staticcheck** running with gometalinter as there is an issue with it, see [#369](https://github.com/golangci/golangci-lint/issues/369).

IMHO, this add value to the linter checks, as it is running quickly and smartly.

See, i.e. running **all** linters except staticcheck takes 11s!!

```
$ time golangci-lint run --print-issued-lines=false --enable-all --deadline 10m --disable=staticcheck | awk '{print $NF}'| sort | uniq -c
     50 (dupl)
     50 (errcheck)
     50 (gochecknoglobals)
      3 (gochecknoinits)
     40 (gocritic)
      7 (gocyclo)
      3 (goimports)
     50 (golint)
     20 (gosec)
      4 (interfacer)
     50 (lll)
     28 (maligned)
      5 (misspell)
     11 (prealloc)
     15 (scopelint)
      3 (stylecheck)
      2 (typecheck)
     50 (unparam)

real	0m11.028s
user	0m31.194s
sys	0m2.745s
```

And here what we would be running with this PR,

```
$ time golangci-lint run --deadline 10m --disable-all --enable=goconst --enable=deadcode --enable=gofmt --enable=ineffassign --enable=structcheck --enable=unconvert --enable=varcheck

real	0m3.134s
user	0m4.476s
sys	0m0.889s
```

### CODE REVIEW

File: pkg/api/alerting_test.go [hunk 1]
```diff
@@ -66,7 +66,7 @@ func TestAlertingApiEndpoint(t *testing.T) {
 			})
 		})
 
-		loggedInUserScenarioWithRole("When calling GET on", "GET", "/api/alerts?dashboardId=1", "/api/alerts", m.ROLE_EDITOR, func(sc *scenarioContext) {
+		loggedInUserScenarioWithRole("When calling GET on", getMethod, "/api/alerts?dashboardId=1", "/api/alerts", m.ROLE_EDITOR, func(sc *scenarioContext) {
```
Comment (bergquist) [inline pkg/api/alerting_test.go]: Please use `http.MethodGet` instead of a custom variable.

File: .circleci/config.yml [hunk 2]
```diff
@@ -82,8 +82,8 @@ jobs:
     steps:
       - checkout
       - run:
-          name: Gometalinter tests
-          command: './scripts/gometalinter.sh'
+          name: linters tests
```
Comment (bergquist) [inline .circleci/config.yml]: `linters` might be a bit confusing since we have both backend and frontend linters.

Comment (bergquist) [reply .circleci/config.yml]: How about backendlinters?

**Comment by torkelo** [comment]
Can you update this? looks close to mergable

**Comment by mjtrangoni** [comment]
@torkelo sure, sorry for the delay.

File: .circleci/config.yml (hunk 2 above)
Comment (mjtrangoni) [reply .circleci/config.yml]: @bergquist Good point. I renamed it to `linters-backend`, similar to `test-frontend` and `test-backend`.

File: pkg/api/alerting_test.go (hunk 1 above)
Comment (mjtrangoni) [reply pkg/api/alerting_test.go]: Good catch! Thanks!

**Comment by mjtrangoni** [comment]
@torkelo done, and rebased against master branch. I see this `go vet` issue, which makes the CI to fail local.

```
$ go vet ./pkg/...
# github.com/grafana/grafana/pkg/models
pkg/models/dashboard_version.go:36:2: struct field CreatedBy repeats json tag "createdBy" also at dashboard_version.go:25
```

**Comment by torkelo** [comment]
that looks like a bug, bad code to have duplicate property name like that.

**Comment by mjtrangoni** [comment]
@torkelo This is working now, and with `go1.12` vet.

**Comment by bergquist** [comment]
We dont use 1.12 yet since `goconvey` is broken in 1.12

**Comment by mjtrangoni** [comment]
@bergquist Thanks! @torkelo This would be ready to merge then.

**Comment by bergquist** [comment]
I dont like those `goconst` changes. 
replacing all strings like that I tests makes it harder to understand the code. Any chance you can revert those changes? except that I think we should merge this.

**Comment by bergquist** [comment]
Big thank you for taking the time to contribute to Grafana and Im sorry my feedback in this case was annoying since I asked you to change back again. Ive partially replaced gometalinter with golangci-lint in this pr https://github.com/grafana/grafana/pull/16610 I created a new PR since this was getting way behind master. Im closing this PR since its not even further away from master.

Once again.... sorry for the bad feedback.

**Comment by mjtrangoni** [comment]
Hi @bergquist,
No problem, I had less spare time last months. But I am always happy to contribute.

**Pull request closed without merging**